### PR TITLE
hubble: Support `--since` requests in combination with follow-mode

### LINKED ISSUE #363: GetFlows/hubble observe doesn't support both --since and --follow at the same time
I'm using hubble observe as an example, however I believe it's a problem inside GetFlows implementation, and not on a client, since hubble-ui experiencing similar issue with GetFlows API.

Currently, it's not possible to do the following command:

```
hubble observe --since=1m --follow
```

Expected behavior for this command would be to first return last 1m of data from buffer, than continue streaming flows to the client. However, this is not happening, it's only streaming new flows.

Compare this with --last behavior:

```
hubble observe --last=1000 --follow
```

That command returns last 1000 flows, than continues streaming new flows. 

hubble-ui's use case for having both --follow and --since=1m is simple: we want to show user a service map, based on the last X minutes of buffered flows, instead of showing them an empty screen and making them wait for new flows to arrive.

### PR DESCRIPTION
**Pull request by gandro**
Previously, the observer implementation assumed time range filters on
the request are not compatible with follow-mode. This however is no
longer the case, we can now apply the since filter when rewinding the
ring buffer. This means that if the user specifies a `since` timestamp,
we first dump all flows newer than `since` before we enter follow-mode.

Fixes: cilium/hubble#363

```release-note
Fix issue where Hubble did not properly support `--follow` queries with a `--since` filter
```

### CODE REVIEW

**Review by rolinh — APPROVED** [review]
That's a nice improvement and the code is even more intuitive to read!

**Review by michi-covalent — APPROVED** [review]
💯

File: pkg/hubble/observer/local_observer.go
Comment (glibsm) [inline pkg/hubble/observer/local_observer.go]: `until 24h ago` is a valid rewind scenario, no?

Comment (gandro) [reply pkg/hubble/observer/local_observer.go]: Yes, good catch. I don't think the CLI supports `until`+`follow` right now, but no reason to not support it server-side.

Comment (glibsm) [reply pkg/hubble/observer/local_observer.go]: I don't think `until`+`follow` is a very common use-case. I think I misread the code, I was just trying to make sure that `until 24ago no follow` would rewind properly

**Review by glibsm — APPROVED** [review]

**Comment by gandro** [comment]
test-me-please

File: pkg/hubble/observer/local_observer.go
Comment (gandro) [reply pkg/hubble/observer/local_observer.go]: Yes, I also got this wrong just now (thus the force pushes). Turns out we want to rewind (or not rewind) regardless of the value of  `req.Until`. `until` doesn't tell us how much we want to rewind (reading backwards), it only tells us when to stop when reading forward. So I guess the condition was correct in either case.

I believe the server-side implementation as of this PR will actually work with `follow`+`until`, but again this likely needs CLI changes first (as the CLI currently clears until if follow-mode is set) for users to actually be able to use it.

**Comment by gandro** [comment]
test-me-please

**Review by christarazi — APPROVED** [review]
net-next hit https://github.com/cilium/cilium/issues/13062. All other tests have passed and have all code owner approval. Merging.

**Pull request merged**
